pkg/indexers/indexers.go

```diff
@@ -92,3 +96,18 @@ func IndexBySyncerFinalizerKey(obj interface{}) ([]string, error) {
 
 	return syncerFinalizers, nil
 }
+
+// ByIndex returns all instances of T that match indexValue in indexName in indexer.
+func ByIndex[T runtime.Object](indexer cache.Indexer, indexName, indexValue string) ([]T, error) {
+	list, err := indexer.ByIndex(indexName, indexValue)
+	if err != nil {
+		return nil, err
+	}
+
+	var ret []T
+	for _, o := range list {
+		ret = append(ret, o.(T))
```

Comment:
since `cache.Indexer` is not (yet) parameterized, this cast can fail - is that different from the preexisting behavior?

Reply:
Same as preexisting behavior. I can put in a check if you think we need it?

Reply:
If it's the same, we don't